arbnode/dataposter/redis/redisstorage.go

```diff
@@ -144,21 +142,20 @@ func (s *Storage) Put(ctx context.Context, index uint64, prev, new *storage.Queu
 			if err != nil {
 				return fmt.Errorf("failed to validate item already in redis at index%v: %w", index, err)
 			}
-			prevItemEncoded, err := rlp.EncodeToBytes(prev)
+			prevItemEncoded, err := s.encDec().Encode(prev)
 			if err != nil {
 				return err
 			}
 			if !bytes.Equal(verifiedItem, prevItemEncoded) {
 				return fmt.Errorf("%w: replacing different item than expected at index %v", storage.ErrStorageRace, index)
```

Comment:
This will incorrectly trigger on the upgrade from legacy format to new format. Maybe we could add an Equals method to the EncoderDecoder or something?

Reply:
Great catch! Added normalization of encoding (decode/encode) to make sure it's using same encoding. We can drop that part after we finally migrate.